install-dev/upgrade/upgrade.php

```diff
@@ -110,7 +110,8 @@
 }
 
 $result = '<?xml version="1.0" encoding="UTF-8"?>';
-if (empty($upgrade->hasFailure())) {
+$hasFailure = $upgrade->hasFailure();
```

Comment:
It depends if you want a contribution on this. If you put the change in your PR, I will merge it. Otherwise I wil open another one. :wink: